File: rust/src/models/entitlement_product_ref.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::product_id::ProductId;

/// Minimal reference to the product a feature belongs to.
#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct EntitlementProductRef {
    pub id: ProductId,

    pub name: String,
}

impl EntitlementProductRef {
    pub fn new(id: ProductId, name: String) -> Self {
        Self { id, name }
    }
}

File: typescript/src/models/entitlementProductRef.ts
// this file is @generated
import { type ProductId, ProductIdSerializer } from "./productId";
/** Minimal reference to the product a feature belongs to. */
export interface EntitlementProductRef {
  id: ProductId;

  name: string;
}

export const EntitlementProductRefSerializer = {
  _fromJsonObject(object: any): EntitlementProductRef {
    return {
      id: ProductIdSerializer._fromJsonObject(object["id"]),
      name: object["name"],
    };
  },

  _toJsonObject(self: EntitlementProductRef): any {
    return {
      id: ProductIdSerializer._toJsonObject(self.id),
      name: self.name,
    };
  },
};
